pavelib/quality.py

```diff
@@ -258,6 +264,16 @@ def run_quality(options):
     pep8_reports = u' '.join(pep8_files)
 
     try:
+        violations_str = None
+        with open(pep8_reports, 'r') as f:
+            violations_list = f.readlines()
+            violations_str = '\n'.join(violations_list)
+
+        if violations_str:
+            raise BuildFailure(
+                'Number of pep8 violations should be 0; found these violations:\n' + violations_str
+            )
+
         sh(
             "diff-quality --violations=pep8 {pep8_reports} {percentage_string} "
```

Comment:
Yeah, that's exactly my goal. I had some stuff locally but it was _totally_ wrong. so I need to backtrack and rethink how I'm doing this. That's why I raised the idea of adding a flag to diff-quality itself that would just print all violations regardless of if they're in the diff or not (that information is gathered, iirc)